From 391ba82d8975f7c9fbd35edeecbd7ca96942b3bd Mon Sep 17 00:00:00 2001
From: Alison Wang <alison.wang@nxp.com>
Date: Tue, 9 Oct 2018 16:47:29 +0800
Subject: [PATCH 340/706] drm: cnd_hdp: Fix the wrong calculation for
 DP_HORIZONTAL_ADDR_Param

This patch fixs the wrong calculation for DP_HORIZONTAL_ADDR_Param, the
value of hsync should be mode->hsync_end - mode->hsync_start.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
(cherry picked from commit 3035a83b277d1c9a377c6656172889c1fcc8bf69)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/imx/hdp/cdn_hdp/API_DPTX.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/hdp/cdn_hdp/API_DPTX.c b/drivers/gpu/drm/imx/hdp/cdn_hdp/API_DPTX.c
index 0894e446daa6..dc5d3ea87b60 100644
--- a/drivers/gpu/drm/imx/hdp/cdn_hdp/API_DPTX.c
+++ b/drivers/gpu/drm/imx/hdp/cdn_hdp/API_DPTX.c
@@ -516,7 +516,8 @@ CDN_API_STATUS CDN_API_DPTX_Set_VIC(state_struct *state,
 		((mode->flags & DRM_MODE_FLAG_INTERLACE ?
 		  mode->vdisplay / 2 : mode->vdisplay) << 16);
 
-	DP_HORIZONTAL_ADDR_Param = (mode->hdisplay << 16) + mode->hsync;
+	DP_HORIZONTAL_ADDR_Param = (mode->hdisplay << 16) + mode->hsync_end -
+		mode->hsync_start;
 
 	DP_VERTICAL_0_ADDR_Param =
 	    (mode->flags & DRM_MODE_FLAG_INTERLACE ?
-- 
2.17.1

